src/lib/validate-input.js

```diff
@@ -662,6 +662,68 @@ export function validateScatterElementsParams(input, indices, updates, {axis = 0
   }
 }
 
+export function validateScatterNDParams(input, indices, updates) {
+  // Refer to https://onnx.ai/onnx/operators/onnx__ScatterND.html
+  // ScatterND takes three inputs data tensor of rank r >= 1, indices tensor of rank q >= 1,
+  // and updates tensor of rank q + r - indices.shape[-1] - 1.
+
+  const inputRank = input.rank;
+  if (inputRank < 1) {
+    throw new Error(`The input should be at least a 1-D tensor.`);
+  }
+
+  const indicesRank = indices.rank;
+  if (indicesRank < 1) {
+    throw new Error(`The indices should be at least a 1-D tensor.`);
+  }
+
+  const indicesShape = indices.shape;
+  const lastIndicesSize = indicesShape[indicesRank - 1];
+  if (lastIndicesSize < 1 || lastIndicesSize > inputRank) {
+    throw new Error(`The indices.shape[-1] should be in the range [1, ${inputRank}].`);
+  }
+
+  const inputShape = input.shape;
+  const indicesTotal = sizeOfShape(indicesShape);
+  const updatedLocationSet = new Set();
+  for (let indicesIndex = 0; indicesIndex < indicesTotal; indicesIndex += lastIndicesSize) {
+    const originIndicesArray = [];
+    const indicesArray = [];
+    for (let i = 0; i < lastIndicesSize; i++) {
+      const indicesValue = indices.getValueByIndex(indicesIndex + i);
+      const maxSize = inputShape[i];
+      if (!Number.isInteger(indicesValue) ||
+        indicesValue < -maxSize ||
+        indicesValue > maxSize - 1) {
+        throw new Error(`Invalid indices value - it should be an integer in the interval ` +
```

Comment:
Using comma seems more consistent and use term "range". Could we also log the invalid value? Something like
```suggestion
        throw new Error(`Invalid indices value (${indicesValue}), it should be an integer in the range ` +
```